python/GafferUI/CompoundEditor.py

```diff
@@ -300,20 +308,31 @@ def __restoreEditorState( self, editorState ) :
 
 		for path, state in editorState.items() :
 
-			if "driver" in state :
-				editor = self.__editorAtPath( path )
-				driver = self.__editorAtPath( state["driver"] )
-				# The mode may not be registered any more, so make sure
-				# we fail gracefully here
-				try :
+			editor = self.__editorAtPath( path )
+
+			try :
+
+				if "driver" in state :
+
+					driver = self.__editorAtPath( state["driver"] )
 					editor.setNodeSetDriver( driver, state["driverMode"] )
-				except Exception as e :
-					sys.stderr.write(
-						"Unable to restore node set driver for {editor}: {error}\n".format(
-							editor = path,
-							error = "%s: %s" % ( type(e), e )
-						)
+
+				elif "nodeSet" in state and isinstance( editor, GafferUI.NodeSetEditor ) :
```

Comment:
Why `isinstance( editor, GafferUI.NodeSetEditor )`? Isn't it impossible to serialise `nodeSet` for anything but a `NodeSetEditor`?

Reply:
It is indeed. Good post, thanks. https://github.com/GafferHQ/gaffer/pull/3360/commits/45bc63ed9e4b00322775a27def17db7bb41f95c8